tests/resources/gateway_control_plane_test.go

```diff
@@ -0,0 +1,78 @@
+package tests
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/config"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/plancheck"
+)
+
+func TestGatewayControlPlane(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		testName    string
+		name        string
+		clusterType string
+	}{
+		{
+			testName:    "hybrid",
+			name:        "Terraform Control Plane",
+			clusterType: "CLUSTER_TYPE_CONTROL_PLANE",
+		},
+		{
+			testName:    "ingress controller",
+			name:        "KIC Terraform Control Plane",
+			clusterType: "CLUSTER_TYPE_K8S_INGRESS_CONTROLLER",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run("type "+tc.testName, func(t *testing.T) {
+			resource.Test(t, resource.TestCase{
+				ProtoV6ProviderFactories: providerFactory,
+				Steps: []resource.TestStep{
+					{
+						Config:          providerConfigUs,
+						ConfigDirectory: config.TestNameDirectory(),
+						Check: resource.ComposeAggregateTestCheckFunc(
+							resource.TestCheckResourceAttr("konnect_gateway_control_plane.tfdemo", "name", tc.name),
+							resource.TestCheckResourceAttr("konnect_gateway_control_plane.tfdemo", "cluster_type", tc.clusterType),
+						),
+					},
+					{
+						Config:          providerConfigUs,
+						ConfigDirectory: config.TestNameDirectory(),
```

Comment:
@mheap  how is this working since the docs say:

```
	// When running Terraform operations for the test, Terraform will
	// be executed with copies of the files of this directory as its
	// working directory. Only one of Config, ConfigDirectory or
	// ConfigFile can be set otherwise an error will be returned.
```

?

Reply:
I have no idea 😁

But it does work to inject the provider config using `Config` and provide test files using `ConfigDirectory`